conans/test/functional/command/profile_test.py

```diff
@@ -128,3 +128,26 @@ def test_profile_new(self):
         assert "MyProfile2' already exists" in c.out
 
         c.run("profile detect --name=./MyProfile2 --force")  # will not raise error
+    
+    @pytest.mark.skipif(platform.system() != "Windows", reason="Requires Windows and msvc")
+    def test_profile_new_msvc_vcvars(self):
+        c = TestClient()
+
+        # extract location of cl.exe from vcvars
+        vcvars = vcvars_command(version="15", architecture="x64")
+        ret = c.run_command(f"{vcvars} && where cl.exe")
+        assert ret == 0
+        cl_executable = c.out.splitlines()[-1]
+        assert os.path.isfile(cl_executable)
+        cl_location = os.path.dirname(cl_executable)
+        
+        # Try different variations, including full path and full path with quotes 
+        for var in ["cl", "cl.exe", cl_executable, f'"{cl_executable}"']:
+            output = RedirectedTestOutput()
+            with redirect_output(output):
+                with environment_update({"CC": var, "PATH": cl_location}):
+                    result = detect_defaults_settings()
```

Comment:
It this is a functional test, maybe do we want to run ``client.run("profile detect")`` instead?

Reply:
I considered that - but I need to be able to influence the environment the `TestClient`  is exposed to with at least two environment variables, do we have convenience functionality for this, or would influencing via `os.environ` be sufficient? This test is similar to the `test_detect_default_in_mac_os_using_gcc_as_default` test above, which also needs environment variables.

Reply:
I think that running ``client.run()`` under that same ``environment_update()`` should have the same behavior?

Reply:
Done!